src/integrationTest/java/org/opensearch/security/SearchOperationTest.java

```diff
@@ -371,6 +373,9 @@ public class SearchOperationTest {
             USER_ALLOWED_TO_PERFORM_INDEX_OPERATIONS_ON_SELECTED_INDICES,
             USER_ALLOWED_TO_CREATE_INDEX
         )
+        .nodeSettings(
+            Map.of(FeatureFlags.TELEMETRY_SETTING.getKey(), true, TelemetrySettings.TRACER_FEATURE_ENABLED_SETTING.getKey(), true)
```

Comment:
@peternied, As mentioned by @reta these settings are not going to work and tracing would be noop in this scenario. In case of NoopTracer we tried to implement in a way that its shouldn't alter the execution pattern by unnecessarily introducing the tracer related core changes (Tracing being an experimental feature eg. is TraceableHttpChannel here. If tracer is noop we are not wrapping the HttpChannel hence its working with FeatureFlag disabled).

Now to test the telemetry plugin and security plugin we need to install them together and execute the tests.

To validate the telemetry working we need to have some search/indexing requests and validate if traces are being generated and which is not very straight forward.

Reply:
>To validate the telemetry working we need to have some search/indexing requests and validate if traces are being generated and which is not very straight forward.

Reply:
How are we testing security with other plugins today?

Reply:
One more thing, Even if we write the integration test, it's still going to fail bcauase the issue is partially fixed and the check inside NettyAttribute.java is still there - 

```
public static <T> Optional<T> popFrom(final RestRequest request, final AttributeKey<T> attribute) {
        if (request.getHttpChannel() instanceof Netty4HttpChannel) {
            Channel nettyChannel = ((Netty4HttpChannel) request.getHttpChannel()).getNettyChannel();
            return Optional.ofNullable(nettyChannel.attr(attribute).getAndSet(null));
        }
        return Optional.empty();
    }
```

Reply:
Ah, correct, this part is on me, thanks for reminding @Gaganjuneja

Reply:
@Gaganjuneja You can see an example of how anomaly-detection tests with security here: https://github.com/opensearch-project/anomaly-detection/blob/main/.github/workflows/test_security.yml

They run their full integ test suite in a cluster w/ the security plugin installed.

Reply:
> One more thing, Even if we write the integration test, it's still going to fail bcauase the issue is partially fixed and the check inside NettyAttribute.java is still there -
> 
> ```
> public static <T> Optional<T> popFrom(final RestRequest request, final AttributeKey<T> attribute) {
>         if (request.getHttpChannel() instanceof Netty4HttpChannel) {
>             Channel nettyChannel = ((Netty4HttpChannel) request.getHttpChannel()).getNettyChannel();
>             return Optional.ofNullable(nettyChannel.attr(attribute).getAndSet(null));
>         }
>         return Optional.empty();
>     }
> ```



> Ah, correct, this part is on me, thanks for reminding @Gaganjuneja

@reta, What is your opinion, this should be addressed here on in another PR. 

@DarshitChanpura, Once the above issue is addressed, I will write the integration test.

Reply:
> @reta, What is your opinion, this should be addressed here on in another PR.

Up to you @Gaganjuneja